spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/connection/DefaultTcpNioSSLConnectionSupport.java

```diff
@@ -46,18 +46,30 @@ public DefaultTcpNioSSLConnectionSupport(TcpSSLContextSupport sslContextSupport)
 	/**
 	 * Creates a {@link TcpNioSSLConnection}.
 	 */
+	@Override
 	public TcpNioConnection createNewConnection(SocketChannel socketChannel, boolean server, boolean lookupHost,
 			ApplicationEventPublisher applicationEventPublisher, String connectionFactoryName) throws Exception {
 		SSLEngine sslEngine = this.sslContext.createSSLEngine();
+		postProcessSSLEngine(sslEngine);
 		TcpNioSSLConnection tcpNioSSLConnection = new TcpNioSSLConnection(socketChannel, server, lookupHost,
 				applicationEventPublisher, connectionFactoryName, sslEngine);
 		tcpNioSSLConnection.init();
 		return tcpNioSSLConnection;
 	}
 
+	@Override
 	public void afterPropertiesSet() throws Exception {
 		this.sslContext = this.sslContextSupport.getSSLContext();
```

Comment:
Not related to this fix, but can't we get rid of this extra `InitializingBean` operation and just `getSSLContext()` directly in the ctor ?